### PR TITLE
feat: Import integration in draft status.

### LINKED ISSUE #109: Import integration in draft status
|</img> @chirino | [2017-11-06](https://github.com/syndesisio/syndesis-rest/issues/779) |
|-|-|

### PR DESCRIPTION
**Pull request by chirino**
Fixes #109.

### CODE REVIEW

File: rest/rest/src/main/java/io/syndesis/rest/v1/handler/integration/IntegrationHandler.java [hunk 1]
```diff
@@ -197,6 +197,7 @@ public Integration create(@Context SecurityContext sec, @ConvertGroup(from = Def
             .createdDate(rightNow)
             .currentStatus(determineCurrentStatus(encryptedIntegration))
             .userId(sec.getUserPrincipal().getName())
+            .stepsDone(new ArrayList<>())
```
Comment (rhuss) [inline rest/rest/src/main/java/io/syndesis/rest/v1/handler/integration/IntegrationHandler.java]: Shouldn't this be the default in the constructor of an integration ?

File: rest/rest/src/main/java/io/syndesis/rest/v1/handler/integration/IntegrationSupportHandler.java [hunk 2]
```diff
@@ -115,11 +120,18 @@ public int importModels(ModelExport export) throws IOException {
...
+                    integration = new Integration.Builder()
+                        .createFrom(integration)
+                        .desiredStatus(Integration.Status.Draft)
+                        .build();
+
                     // Do we need to create it?
                     if (dataManager.fetch(Integration.class, integration.getId().get()) == null) {
-                        dataManager.create(integration);
+                        LOG.info("Creating integration: "+integration.getName());
+                        integrationHandler.create(sec, integration);
```
Comment (rhuss) [inline rest/rest/src/main/java/io/syndesis/rest/v1/handler/integration/IntegrationSupportHandler.java]: Architectural wise I think it's not so nice to call a Rest endpoint from another rest endpoint as this hinders the API to evolve freely and introduces an unnecessary coupling. Instead, I would extract the common used code into a lower-layer class and use this from both, here and the endpoint.

File: rest/rest/src/main/java/io/syndesis/rest/v1/handler/integration/IntegrationHandler.java (hunk 1 above)
Comment (chirino) [reply rest/rest/src/main/java/io/syndesis/rest/v1/handler/integration/IntegrationHandler.java]: Not the case when we are creating an integration from an import.

Comment (chirino) [reply rest/rest/src/main/java/io/syndesis/rest/v1/handler/integration/IntegrationHandler.java]: Not the case when we are creating an integration from an import.

**Pull request merged**